v5e_high_performance.md

```diff
@@ -0,0 +1,139 @@
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ -->
+
+# High performance 16B, 32B, 64B, and 128B models on TPU v5e
+The following are details on how to achieve up to 66.86% MFU results on TPU v5e for 16B, 32B, 64B, and 128B parameter model configurations. 
+
+| Model Size | Hardware    | TFLOP/sec/chip   |   MFU  |
+| ---------- | ----------- | ---------------- | ------ |
+| 16B        | 1x v5e-256  | 120              | 61.10% |
+| 16B        | 2x v5e-256  | 117              | 59.37% |
+| 32B        | 1x v5e-256  | 132              | 66.86% |
+| 32B        | 2x v5e-256  | 128              | 64.81% |
+| 64B        | 1x v5e-256  | 118              | 59.90% |
+| 64B        | 2x v5e-256  | 112              | 56.66% |
+| 128B       | 1x v5e-256  | 110              | 56.06% |
+| 128B       | 2x v5e-256  | 110              | 55.82% |
+
+We have shown full scaling performance results up to 16 v5e pods in [README.md](https://github.com/google/maxtext/blob/main/README.md#tpu-v5e
+) for these configurations, however they are capable of scaling up to hundreds of v5e pods. The full configurations and scripts can be found in [MaxText/configs/largest_job/](https://github.com/google/maxtext/tree/main/MaxText/configs/largest_job). 
+
+Note that these configurations do 3 things:
+* Sets various XLA compiler flags as `LIBTPU_INIT_ARGS` to optimize runtime performance.
+* Runs [rto_setup.sh](https://github.com/google/maxtext/blob/main/rto_setup.sh) or [gke_rto_setup.sh](https://github.com/google/maxtext/blob/main/gke_rto_setup.sh) to optimize communication protocols for network performance. 
+(This only needs to be run once on each worker)
+* Runs [train.py](https://github.com/google/maxtext/blob/main/MaxText/train.py) with specific hyper-parameters (batch size, etc.)
+
+
+## Reproduction Instructions
+Depending on your organization's set up, these instructions may vary. Here we provide instructions for running on a vanilla GCE setup or a GKE with XPK setup. Feel free to reach out with questions if your organization has a unique setup.
+
+### Running on GCE
+1. Create a custom MTU network to optimize network performance and give it firewall rules. If you are unable to complete this step, you may skip it. This step is not necessary, however performance degradation from scaling up the number of v5e pods over DCN will become more noticable.
+     
+     Create a network with an MTU of 8896 bytes and set up firewall rules. (Creating a network requires `compute.networks.create` permission in your project)
+     ```
+     gcloud compute networks create mtu9k --mtu=8896 --project={PROJECT} --subnet-mode=auto --bgp-routing-mode=regional
```

Comment:
nit $ in front of {PROJECT} + other PROJECT variable on line 52 and incorrect $ position on line 58 for PROJECT

Reply:
Good catch! Thank you